File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution(object):
    def numberOfSubarrays(self, nums, k):
     odd=0
     d={0:1}
     c=0
     for i in nums:
        if (i%2!=0):
            odd+=1
        if odd-k in d:
            c+=d[odd-k]
        d[odd]=d.get(odd,0)+1
     return c

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
       HashMap<Integer,Integer> freq=new HashMap<>();
       freq.put(0,1);
       int sum=0;
       int c=0;
       for(int i=0;i<nums.length;i++){
        if(nums[i]%2!=0) sum+=1;
        if(freq.containsKey(sum-k)) c+=freq.get(sum-k);
        freq.put(sum,freq.getOrDefault(sum,0)+1);
       }
    return c;
    }
}